ksqldb-common/src/test/java/io/confluent/ksql/function/UdfIndexTest.java

```diff
@@ -231,6 +238,127 @@ public void shouldChooseCorrectMap() {
     assertThat(fun.name(), equalTo(EXPECTED));
   }
 
+  @Test
+  public void shouldChooseCorrectLambdaFunction() {
```

Comment:
this is kind-of a weird test - we're testing the following signature:
```java
public String foo(List<T> list, Function<A, B> func);
```
by calling
```java
foo(list_of_ints, a_string -> an_int)
```

While there's nothing preventing that from working, I feel like we should be testing the case where input generic of the function should match the generic of the list. To make sure that it's working, we probably want to test that it chooses the right one if we do something more complicated:

```java
public String foo(Map<K, V> map, Function<K, K2> keyMapper);
public String foo(Map<K, V> map, Function<V, V2> valueMapper);
```

If I pass in `MAP<STRING, INTEGER>` and `LAMBDA<[STRING], STRING>` it should give me the first, but if I pass in `LAMBDA<[INTEGER], INTEGER>` it should give me the second.

Reply:
Just updated (cc @stevenpyzhang )